src/csharp/Microsoft.Spark/Sql/RowConstructor.cs

```diff
@@ -65,15 +65,6 @@ public object construct(object[] args)
                 s_schemaCache = new Dictionary<string, StructType>();
             }
 
-            // When a row is ready to be materialized, then construct() is called
-            // on the RowConstructor which represents the row.
-            if ((args.Length == 1) && (args[0] is RowConstructor rowConstructor))
-            {
-                // Construct the Row and return args containing the Row.
-                args[0] = rowConstructor.GetRow();
-                return args;
-            }
-
```

Comment:
Can you also get rid of the loop in the .GetRow() method.  Move the materialization of the inner rows to the Row.Convert() method.

Reply:
Why is that?

Reply:
This should save us from looping through the columns of the Row twice.  Since Row.Convert is already looping through the field types, it should be okay to do the conversion there.

Reply:
I would like to contain the `RowConstructor` calls as much as possible (we already had few bugs around this). Also, it's much easier to understand and consistent now that `RowConstructor` is used close to  `PythonSerDe.GetUnpickledObjects`, one in `RowCollector` and one in `PicklingSqlCommandExecutor`.

If looping through columns twice is a concern (I would think it would be dominated by materialization anyway and we already have a lot of room for improvements on that code path), we can do that as a separate PR.

Reply:
Sure we can leave as is then.